### PR TITLE
first part was implemented

### PR DESCRIPTION
**Pull request by Mandybur800**

### CODE REVIEW

**Review by Max-Pochepets — APPROVED** [review]

File: pom.xml [hunk 1]
```diff
@@ -9,8 +9,51 @@
...
+                    <configuration>
+                        <source>${jdk.version}</source>
+                        <target>${jdk.version}</target>
+                        <encoding>${project.build.sourceEncoding}</encoding>
+                    </configuration>
+                </plugin>
+            </plugins>
+        </pluginManagement>
+    </build>
 </project>
```
Comment (Sofasmile) [inline pom.xml]: let's add empty line at the end of the file

File: src/main/java/com/manufacturing/dao/impl/ManufacturerDaoImpl.java [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+                .filter(m -> Objects.equals(m.getId(), manufacturerId))
+                .findFirst();
+    }
+
+    @Override
+    public Manufacturer update(Manufacturer manufacturerUpdate) {
+        Manufacturer manufacturer = getAllManufacturers().stream()
+                .filter(m -> Objects.equals(m.getId(), manufacturerUpdate.getId()))
+                .findFirst().get();
+        manufacturer.setName(manufacturerUpdate.getName());
```
Comment (Sofasmile) [inline src/main/java/com/manufacturing/dao/impl/ManufacturerDaoImpl.java]: this code isn't flexible. We should change the logic of this method if the manufacturer object is changed

let's use the method `set` from the list for replacing an object

File: src/main/java/com/manufacturing/dao/impl/ManufacturerDaoImpl.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+        manufacturer.setCountry(manufacturerUpdate.getCountry());
+        return manufacturerUpdate;
+    }
+
+    @Override
+    public boolean deleteById(Long manufacturerId) {
+        for (Manufacturer manufacturer: getAllManufacturers()) {
+            if (Objects.equals(manufacturer.getId(), manufacturerId)) {
+                return Storage.manufacturers.remove(manufacturer);
```
Comment (Sofasmile) [inline src/main/java/com/manufacturing/dao/impl/ManufacturerDaoImpl.java]: we can use method `removeIf()`

File: src/main/java/com/manufacturing/service/impl/ManufacturerServiceImpl.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+import com.manufacturing.lib.Inject;
+import com.manufacturing.lib.Service;
+import com.manufacturing.model.Manufacturer;
+import com.manufacturing.service.ManufacturerService;
+import java.util.List;
+
+@Service
+public class ManufacturerServiceImpl implements ManufacturerService {
+    @Inject
+    ManufacturerDao manufacturerDao;
```
Comment (Sofasmile) [inline src/main/java/com/manufacturing/service/impl/ManufacturerServiceImpl.java]: let's add access modifier)

File: src/main/java/com/manufacturing/dao/ManufacturerDao.java [hunk 5]
```diff
@@ -0,0 +1,19 @@
...
+
+    Optional<Manufacturer> get(Long manufacturerId);
+
+    Manufacturer update(Manufacturer manufacturer);
+
+    boolean deleteById(Long manufacturerId);
+
+    boolean delete(Manufacturer manufacturer);
+
+    List<Manufacturer> getAllManufacturers();
```
Comment (Sofasmile) [inline src/main/java/com/manufacturing/dao/ManufacturerDao.java]:
```suggestion
    List<Manufacturer> getAll();
```

**Review by Sofasmile — CHANGES_REQUESTED** [review]
good job)

**Review by Sofasmile — APPROVED** [review]

**Pull request merged**